storage/util/dynamic_percentile.go

```diff
@@ -0,0 +1,157 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package util
+
+import (
+	"fmt"
+	"math"
+	"sync"
+)
+
+// MetricType is the metric type DynamicPercentile algo expects.
+type MetricType int64
```

Comment:
Do we need an aliased type for this? Seems like just leaving all values as an int64 would be fine. Also, what type does the internal version of this use for the values? Wondering if we might have issues with precision depending on what values we choose.

Reply:
Also, we could just leave this as a time.Duration and I could pipe through throughput as "the avg duration to transfer 1 MiB" for example (or 1 GiB).

Reply:
Internal version uses time.Duration only. 

Sure, reverting this to duration only. 
`the avg duration to transfer 1 MiB" for example (or 1 GiB).` - This will also work, the only thing we need to handle the stalled stream (0 byte transfer).